src/dynarec/la64/dynarec_la64_660f.c

```diff
@@ -418,6 +493,63 @@ uintptr_t dynarec64_660F(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int
                         VOR_V(q0, q0, d0);
                     }
                     break;
+                case 0x0E:
+                    INST_NAME("PBLENDW Gx, Ex, Ib");
+                    nextop = F8;
+                    GETGX(q0, 1);
+                    GETEX(q1, 0, 1);
+                    u8 = F8;
+                    i32 = 0;
+                    if (q0 != q1) {
+                        if (u8 == 0xff) {
+                            VAND_V(q0, q1, q1);
+                        } else {
+                            /* 64bits */
+                            if ((u8 & 0xf) == 0xf) {
+                                VEXTRINS_D(q0, q1, VEXTRINS_IMM_4_0(0, 0));
+                                u8 &= ~0xf;
+                            }
+                            if ((u8 & 0xf0) == 0xf0) {
+                                VEXTRINS_D(q0, q1, VEXTRINS_IMM_4_0(1, 1));
+                                u8 &= ~0xf0;
+                            }
+                            /* 32bits */
+                            if ((u8 & 0x3) == 0x3) {
+                                VEXTRINS_W(q0, q1, VEXTRINS_IMM_4_0(0, 0));
+                                u8 &= ~0x3;
+                            }
+                            if ((u8 & 0xc) == 0xc) {
+                                VEXTRINS_W(q0, q1, VEXTRINS_IMM_4_0(1, 1));
+                                u8 &= ~0xc;
+                            }
+                            if ((u8 & 0x30) == 0x30) {
+                                VEXTRINS_W(q0, q1, VEXTRINS_IMM_4_0(2, 2));
+                                u8 &= ~0x30;
+                            }
+                            if ((u8 & 0xc0) == 0xc0) {
+                                VEXTRINS_W(q0, q1, VEXTRINS_IMM_4_0(3, 3));
+                                u8 &= ~0xc0;
+                            }
+                            /* 16bits */
+                            if (u8 & 0x1)
+                                VEXTRINS_H(q0, q1, VEXTRINS_IMM_4_0(0, 0));
+                            if (u8 & 0x2)
+                                VEXTRINS_H(q0, q1, VEXTRINS_IMM_4_0(1, 1));
+                            if (u8 & 0x4)
+                                VEXTRINS_H(q0, q1, VEXTRINS_IMM_4_0(2, 2));
+                            if (u8 & 0x8)
+                                VEXTRINS_H(q0, q1, VEXTRINS_IMM_4_0(3, 3));
+                            if (u8 & 0x10)
+                                VEXTRINS_H(q0, q1, VEXTRINS_IMM_4_0(4, 4));
+                            if (u8 & 0x20)
+                                VEXTRINS_H(q0, q1, VEXTRINS_IMM_4_0(5, 5));
+                            if (u8 & 0x40)
+                                VEXTRINS_H(q0, q1, VEXTRINS_IMM_4_0(6, 6));
+                            if (u8 & 0x80)
+                                VEXTRINS_H(q0, q1, VEXTRINS_IMM_4_0(7, 7));
```

Comment:
Ooof, this is really complicated! Are you sure there is no other way?

Reply:
So there is an 8-bit immediate as the "mask", and it's known at compile time. So just write some C code to convert say `0b10010110` into 2 64-bit integers `0xffff00000000ffff` and `0x0000ffffffff0000`, and move them into a vector register? This gives you an element-wise mask, then you can use some vector logical instructions to do the job.

Reply:
On a second thought, with maximum 4 instructions generated, I think the current one is better.